### PR TITLE
refactor: injectable settings for docker use

### LINKED ISSUE #339: Environment variables
we have one build for prod and pre-prod, right now environment variables are "inserted" with `sed` into the settings.json file and index.hml. We need a better / more robust way of handling this. Maybe add perl to NGINX and serve the settings like that?

### PR DESCRIPTION
**Pull request by ia3andy**
For reason see issue #339

BREAKING CHANGES:
- config is now loaded synchronously
- docker image will use settings-tmpl.json in order to inject settings in the frontend, new environment varibales need to be added there.
- new `backend_api_url` config key to use inside services to get the `{backend}/api` url.
- remove `settings.json` and `keycloak.json` (not needed anymore).

closes #339

### CODE REVIEW

**Comment by ia3andy** [comment]
Hold until summit

File: README.md [hunk 1]
```diff
@@ -22,24 +22,21 @@ Start the app by executing the following.
 $ npm start
 ```
 
-If you want the UI to use a local version of the backend, you need to set the following environment variables:
+The frontend will use your localhost backend as default. 
+If you want the UI to use another version of the backend, you need to set the following environment variables:
 
 ```bash   
-$ export LAUNCHER_BACKEND_URL=http://localhost:8080/api/
```
Comment (gastaldi) [inline README.md]: I think it's fine to leave `localhost' here, since it's the most common used copy-pasted snippet

File: Dockerfile.deploy [hunk 2]
```diff
@@ -7,14 +7,18 @@ ENV LANG=en_US.utf8
...
 
 COPY scripts/entrypoint.sh /usr/bin/entrypoint.sh
+COPY scripts/settings-tmpl.json /usr/bin/settings-tmpl.json
 RUN chmod +x /usr/bin/entrypoint.sh
 
 RUN rm -f /usr/share/nginx/html/*
 
 COPY dist /usr/share/nginx/html/launch
 RUN chmod a+w /usr/share/nginx/html/launch
+RUN chmod a+w /usr/share/nginx/html/launch/index.html
```
Comment (gastaldi) [inline Dockerfile.deploy]: isn't there a flag to make `chmod` recursive to the files?

Comment (gastaldi) [reply Dockerfile.deploy]: See Tip 3 in https://hackernoon.com/tips-to-reduce-docker-image-sizes-876095da3b34

Comment (ia3andy) [reply Dockerfile.deploy]: It seems that setting `+w` on `/usr/share/nginx/html/launch` is not needed. I am only keeping `+w` on `/usr/share/nginx/html/launch/index.html` then

**Pull request merged**